version: "3.8"
services:
  faker-db:
    build:
      context: docker-postgres
      dockerfile: Dockerfile
    environment:
      POSTGRES_USER: faker
      POSTGRES_PASSWORD: faker
      POSTGRES_DB: faker
    ports:
      - "5400-5402:5432"
    volumes:
      - faker-volume:/var/lib/postgresql/data
    networks:
      - faker-network
      - de-network
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "faker"]
      interval: 10s
      retries: 3
      start_period: 30s
    restart: always
  pgadmin:
    image: dpage/pgadmin4:7.3
    environment:
      PGADMIN_DEFAULT_EMAIL: faker@faker.com
      PGADMIN_DEFAULT_PASSWORD: password
    ports:
      - "5040:80"
    networks:
      - faker-network
    healthcheck:
      test: ["CMD", "ping localhost:80 || exit 1"]
      interval: 10s
      retries: 3
      start_period: 180s
    restart: always
    depends_on:
      - faker-db
  jupyter-lab:
    build:
      context: docker-jupyter
      dockerfile: Dockerfile
    ports:
      - "8888:8888"
    networks:
      - faker-network
    volumes:
      - .:/projects
    depends_on:
      - faker-db
volumes:
  faker-volume:
networks:
  faker-network:
    external: true
  de-network:
    external: true
